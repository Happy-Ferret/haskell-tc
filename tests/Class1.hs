module Class1 where

data String

class Show a where
  show :: a -> String

class Monad m where
  return :: a -> m a
