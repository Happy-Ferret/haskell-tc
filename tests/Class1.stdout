Bindings:
:: âˆ€ v3 v5. .Monad v3 â‡’ v5 â†’  v3 [v5]
  [4mlist x = return [x][0m
:: v5
  list [4mx[0m = return [x]

Proofs:
coercion: (âˆ€ m a. .Monad m â‡’ a â†’  m a) @ v3 [v5]
  list x = [4mreturn[0m [x]
